### DISCUSSION TITLE
Add multiple points as boundaries

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#21046] [Q&A Meshing] [answered]
Hello everyone, I want to add multiple points (~1k) as boundaries.

I get used to using the ExtraNodesetGenerator to create point boundaries. But it's a huge task for thousands point,

Is there a similar way like ReporterPointSource for adding multiple points (~1k) as boundaries?
https://mooseframework.inl.gov/blackbear/source/dirackernels/ReporterPointSource.html

Thanks a lot,

**Comment by GiudGiud** [comment, marked as answer]
Hello

You're right that it's too much to do manually. Is there anything in common between those points ? Like they are nodes on faces with the same normal? Or they are between two separate subdomains? Or on the outside of known subdomains?

Typically we use these geometric characteristics in the mesh generator system to generate sidesets. Nodesets can be obtained from sidesets with just a parameter to Mesh.

Guillaume

**Reply by Joseph-0123** [reply]
Hello, Guillaume, thanks for your responses. 

Yes. A thousand points set as boundaries are located at the 'Line block' (in red color).  Their interval distance is 10m. They belong to the same block.

Thanks for your help again,

![image](https://user-images.githubusercontent.com/56687392/169072764-7a93b139-40e7-4811-9342-b347c6b3696b.png)

**Reply by GiudGiud** [reply]
Hello

I'd say there's two options :
- make the inner cube a different block (it s totally ok wrt to defining materials and kernels to spread them on two blocks rather than one) then use a sidesetbetweensubdomaingenerator
https://mooseframework.inl.gov/source/meshgenerators/SideSetsBetweenSubdomainsGenerator.html
OR
- use one (or more) parsed expressions to specify the sidesets. Since it's straight it's just  'x<1000.1 & x>999.9 & y>-3000.1' for example for the expression
https://mooseframework.inl.gov/source/meshgenerators/ParsedGenerateSideset.html

Guillaume

**Reply by Joseph-0123** [reply]
Great. They all look useful. Thanks, Guillaume, I will follow their way.

Cheers,
